From cff69cd29ff316a7ac38e7e84db105d0841c5fcb Mon Sep 17 00:00:00 2001
From: Hou Qi <qi.hou@nxp.com>
Date: Thu, 18 Aug 2022 12:20:33 +0800
Subject: [PATCH 11/19] V4L2VDA: fix vp9 crash caused by
 DequeueResolutionChangeEvent

Handle source change event only when decoder receives driver reported
source change event.

Upstream-Status: Inappropriate [NXP specific]
---
 media/gpu/v4l2/legacy/v4l2_video_decode_accelerator.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/media/gpu/v4l2/legacy/v4l2_video_decode_accelerator.cc b/media/gpu/v4l2/legacy/v4l2_video_decode_accelerator.cc
index 89a8976eb457d..e6edbcd97a896 100644
--- a/media/gpu/v4l2/legacy/v4l2_video_decode_accelerator.cc
+++ b/media/gpu/v4l2/legacy/v4l2_video_decode_accelerator.cc
@@ -1276,8 +1276,7 @@ void V4L2VideoDecodeAccelerator::ServiceDeviceTask(bool event_pending) {
     gfx::Size visible_size;
     bool again;
     if (GetFormatInfo(&format, &visible_size, &again) && !again) {
-      resolution_change_pending = true;
-      DequeueResolutionChangeEvent();
+      resolution_change_pending = DequeueResolutionChangeEvent();
     }
   }
 
-- 
2.34.1

